Generals/Code/GameEngine/Source/GameClient/GUI/ControlBar/ControlBar.cpp

```diff
@@ -165,6 +165,19 @@ Player* ControlBar::getCurrentlyViewedPlayer()
 	return ThePlayerList->getLocalPlayer();
 }
 
+Relationship ControlBar::getCurrentlyViewedPlayerRelationship(const Team* team)
+{
+	if (TheControlBar->isObserverControlBarOn())
```

Comment:
I expect this can be simplified to

```cpp
if (Player* player = getCurrentlyViewedPlayer())
  return player->getRelationship(team);

return NEUTRAL;
```